contracts/root/CheckpointManager.sol

```diff
@@ -144,57 +147,42 @@ contract CheckpointManager is ICheckpointManager, Initializable {
         uint256[2] memory message,
         uint256[2] calldata signature,
         bytes calldata bitmap
-    ) private {
+    ) private view {
         uint256 length = currentValidatorSetLength;
         // slither-disable-next-line uninitialized-local
         uint256[4] memory aggPubkey;
-        uint256 firstIndex = 0;
-        bool flag = false;
-        for (uint256 i = 0; i < length; ) {
-            if (_getValueFromBitmap(bitmap, i)) {
-                aggPubkey = currentValidatorSet[i].blsKey;
-                firstIndex = i;
-                flag = true;
-                break;
-            }
-
-            unchecked {
-                ++i;
-            }
-        }
-
-        require(flag, "BITMAP_IS_EMPTY");
-
         uint256 aggVotingPower = 0;
-        for (uint256 i = firstIndex + 1; i < length; ) {
+        for (uint256 i = 0; i < length; ) {
             if (_getValueFromBitmap(bitmap, i)) {
-                uint256[4] memory blsKey = currentValidatorSet[i].blsKey;
-                // slither-disable-next-line calls-loop
-                (aggPubkey[0], aggPubkey[1], aggPubkey[2], aggPubkey[3]) = bn256G2.ecTwistAdd(
-                    aggPubkey[0],
-                    aggPubkey[1],
-                    aggPubkey[2],
-                    aggPubkey[3],
-                    blsKey[0],
-                    blsKey[1],
-                    blsKey[2],
-                    blsKey[3]
-                );
+                if (aggVotingPower == 0) {
```

Comment:
Just thinking out loud, wouldn't be safer checking `aggPubKey.length == 0 `here? Is it feasible doing so? 
Asking because I'm thinking about having validator which (for some strange reason) have `votingPower == 0`.

Reply:
`aggPubKey.length` is always equal to 4, even though it's not explicitly initialized, it's initialized to an array of length 4 populated with zeros

Reply:
I could add an assert statement when setting the new validator set that their voting power is `!= 0`, that should be sufficient